src/qibo/tests/test_core_circuit_noise.py

```diff
@@ -160,3 +160,31 @@ def test_density_matrix_circuit_errors():
     c = Circuit(5)
     with pytest.raises(ValueError):
         c.add(gates.KrausChannel([((0,), np.eye(2))]))
+
+
+def test_circuit_add_sampling():
+    """Check measurements when simulating added circuits with noise"""
+    # Create random noisy circuit and add noiseless inverted circuit
+    gates_set = [gates.X, gates.Y, gates.Z, gates.H, gates.S, gates.SDG, gates.I]
+    circ = Circuit(1)
+    circ_no_noise = Circuit(1)
+    for _ in range(100):
+        new_gate = np.random.choice(gates_set)(0)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
+        circ.add(new_gate)
+        circ_no_noise.add(new_gate)
+        circ.add(gates.PauliNoiseChannel(0, pz=0.01))
+    circ += circ_no_noise.invert()
+    circ.add(gates.M(0))
+
+    # Sampling using 10 shots
+    K.set_seed(123)
+    samples = circ(nshots=10).samples()
+
+    # Sampling using 1 shot in for loop
+    target_samples = []
+    K.set_seed(123)
+    for _ in range(10):
+        target_samples.append(circ(nshots=1).samples())
+
+    K.assert_allclose(np.array(target_samples).flatten(), samples.flatten())
```

Comment:
This fails for:
* tensorflow: `AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'flatten'`
* qibojit-cupy: `TypeError: Implicit conversion to a NumPy array is not allowed. Please use .get() to construct a NumPy array explicitly.`

I believe it is simpler if you just do the indexing manually, eg. something like:
```suggestion
    K.assert_allclose(samples[:, 0], target_samples[:, 0, 0])
```
as `K.assert_allclose` is supposed to properly handle the conversions to numpy for different backends.

Reply:
Perfect, thank you!
I will add your suggestions.